mslib/mscolab/server.py

```diff
@@ -833,10 +833,13 @@ def acs_post_handler():
 
     # Implementation for handling configured SAML assertion consumer endpoints
     for idp_config in setup_saml2_backend.CONFIGURED_IDPS:
-        for assertion_consumer_endpoint in idp_config['idp_data']['assertion_consumer_endpoints']:
-            # Dynamically add the route for the current endpoint
-            APP.add_url_rule(f'/{assertion_consumer_endpoint}/', assertion_consumer_endpoint,
-                             create_acs_post_handler(idp_config), methods=['POST'])
+        try:
+            for assertion_consumer_endpoint in idp_config['idp_data']['assertion_consumer_endpoints']:
+                # Dynamically add the route for the current endpoint
+                APP.add_url_rule(f'/{assertion_consumer_endpoint}/', assertion_consumer_endpoint,
+                                 create_acs_post_handler(idp_config), methods=['POST'])
+        except (NameError, AttributeError, KeyError) as ex:
+            logging.debug("USE_SAML2 is %s, Failure is: %s" % (mscolab_settings.USE_SAML2, ex))
```

Comment:
Two things:
1. I think this should be at least a `logging.warning`, debug messages are not visible by default.
2. The format string is always evaluated, no matter if the log message is emitted or not. To avoid this the logging module uses its own way of string interpolation that should be used. See also: https://stackoverflow.com/questions/34619790/pylint-message-logging-format-interpolation

Reply:
the % , somehow I forgot about.